src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/AStar.java

```diff
@@ -0,0 +1,96 @@
+package edu.greenblitz.tobyDetermined.Nodesssss;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+import java.util.LinkedList;
+
+public class AStar {
+    static LinkedList<NodeArm> open = new LinkedList<>();
+    static LinkedList<NodeArm> closed = new LinkedList<>();
+    static NodeBase nodeBase = NodeBase.getInstance();
+
+   public static double getDistanceFromCurrentNodeToEndOrStartNode(NodeArm point, NodeArm current){
+       double distance =  nodeBase.getDistanceBetweenToPoints(point,current);
+       if(distance < 0){
+           distance*=-1;
+       }
+       return distance;
+   }
+    public static double getDistanceToStartPlusEnd(NodeArm current, NodeArm start, NodeArm end){
+       double gCost = getDistanceFromCurrentNodeToEndOrStartNode(start,current);
+       double hCost = getDistanceFromCurrentNodeToEndOrStartNode(end,current);
+       return gCost+hCost;
+    }
+
+
+    public static NodeArm getLowestFcost(LinkedList<NodeArm> open, NodeArm start, NodeArm end) {
+       if (open.isEmpty()) {
+           return null; // Handle the case where the list is empty
+       }
+       int saveI = 0;
+       double fCost = getDistanceToStartPlusEnd(open.get(0), start, end);
+       for (int i = 1; i < open.size(); i++) {
+           double currentFCost = getDistanceToStartPlusEnd(open.get(i), start, end);
+           if (currentFCost < fCost) {
+               fCost = currentFCost;
+               saveI = i;
+           }
+       }
+       return open.get(saveI);
+   }
+
+
+   public static boolean isInList(NodeArm nodeArm, LinkedList<NodeArm> list){
+       for (NodeArm arm : list) {
+           if (nodeArm.getId() == arm.getId())
+               return true;
+       }
+       return false;
+    }
+
+    public static LinkedList<NodeArm> printPathPath(NodeArm nodeArm){
+       LinkedList<NodeArm>  pathList = new LinkedList<>();
+       LinkedList<NodeArm> getPath = new LinkedList<>();
+       NodeArm current = nodeArm;
+       String returningPath = "";
+        while (current != null) {
+            pathList.add(current);
+            current = current.getParent();
+        }
+
+        for (int i = pathList.size() - 1; i >= 0; i--) {
+            returningPath += pathList.get(i).getId()+", ";
+            getPath.add(pathList.get(i));
+        }
+        returningPath+="\n";
+        SmartDashboard.putString("Returning Path",returningPath);
+        return getPath;
+    }
+    public static LinkedList<NodeArm> getPath(NodeArm start, NodeArm end){
+        open.add(start);
+
+        while (!open.isEmpty()) {
+            NodeArm current = getLowestFcost(open, start, end);
+            open.remove(current);
+            closed.add(current);
+
+            assert current != null;
+            if (current.getId() == end.getId()) {
+                return printPathPath(current);
+            }
+
+            for (NodeArm neighbor : current.getNeighbors()) {
```

Comment:
Are you sure that's what you intended?

Reply:
yes, why do you think that's not?

Reply:
Just looked a bit weird; I don't mind test it before making changes if you really looked at it and sure about it